### ISSUE TITLE
Less: `::before` becomes `: : before` in detached rulesets

### ISSUE THREAD

**Issue by tossp** [prettier/prettier#5791]
`::before` not `: : before`😭

**Prettier 1.16.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABAjgGwLQGcCWAXnLkgAQDkAjAAwAOAHhQDSVUBMjLbALF69QBs-NgA4RFdvSYD2fJgG4AOlBUA6LHigQA7mWAqyZNQDMIAJwAUho2Qw4CxXMxtHgGh0TjHNdfGABrOHNsTABDACM4TBcoWx9PYgTsP0Dg7Hx4AFt9V1sAMgBtABMwmDDsADcwzABXOABeSlRgMOKASWyAXwoyAF0kJCizc29gSFgEGHIKFrbOuCye5TjbLvWbAEoVrpBmEAg6GHxoXGRQMPNzXQAFS4QzlDDKiHxivZAI8zC0mABlOg-fBQADmyBg5nq+wAFjAspgAOrQzIkQFgOB-B6ZfCVTIAT2Q4FwZ32wNwwRgN2+IKyYWQJhq5P2ACtcAwAELfX5-MJZOAAGWBcHpjLgLLZf2BIMwcAAirUIPARZgmSBAeZyeZCTLiR86OZgTAEW8YNDkKIaPt9RByQjvnRCfqSMFKsL9qN0LV8KMqWEaXSkAyVWKQOSsvhwZCQwRQTL5YrhYHRftyhFjcVTch2CnvvhMFKAMIQLK0wnaKBukC1ckAFUijyD5PWQA)
```sh
--parser less
```

**Input:**
```less
@ql-sizes: '10px', '12px', '14px', '16px', '18px', '20px', '24px';

.ql-snow {
  .for(
    @ql-sizes,
    {.ql-size .ql-picker-label,
    .ql-size .ql-picker-item {
      &[data-value= '@{adItem}' ]::before {content: '@{adItem}';
    }}}
  );
}
```

**Output:**
```less
@ql-sizes: "10px", "12px", "14px", "16px", "18px", "20px", "24px";

.ql-snow {
  .for(
    @ql-sizes,
    {.ql-size .ql-picker-label,
    .ql-size .ql-picker-item {&[data-value= "@{adItem}" ]: : before {content: "@{adItem}"
      ;}}}
  );
}

```

**Expected behavior:**
```less
@ql-sizes: '10px', '12px', '14px', '16px', '18px', '20px', '24px';

.ql-snow {
  .for(@ql-sizes, {.ql-size .ql-picker-label,  .ql-size .ql-picker-item {
      &[data-value= '@{adItem}' ]::before {content: '@{adItem}';
    }}}
  );
}
```

**Comment by evilebottnawi** [comment]
Parsed doesn't give information for formating this lines as you provide above, but exists new version of `postcss-less` and you can try update deps

**Comment by thorn0** [comment]
The issue with `::before` is fixed by #6927 

However, detached rulesets are not formatted. Prettier leaves them as is. Tracked in https://github.com/prettier/prettier/issues/4524.
